### PR TITLE
[2.19.x] DDF-6204: Enum forced re-render is now selective

### LINKED ISSUE #6204: Enum component rendering inside Marionette bridge causes flickering
### Description
When a user attempts to exports results from the table view by selecting "Specific Number of Results" and entering a custom value, the modal flashes on each value change due to forced re-render. This happens due to: https://github.com/codice/ddf/blob/fa7ecedae9524982c9aeeae69add30df44cc92ee/ui/packages/catalog-ui-search/src/main/webapp/react-component/enum/enum.tsx#L75

This, however, is necessary in order to maintain functionality for: https://github.com/codice/ddf/issues/5424

### Steps to Reproduce
1. Search to retrieve results (> 0)
2. Open 'Table' view (or add it via 'Add Visual')
3. Click 'Export' button
4. Select 'Specific Number of Results' from the drop-down control
5. Modify/update/delete the pre-filled result input
6. Observe flickering of the modal on each input value modification

#### Expected behavior :
Updating/modifying/deleting the result count does not result in a perceptible flicker

#### Actual behavior:
Updating/modifying/deleting the result count produces flicker in the modal dialog

#### Reproduces how often:
Always

#### Platform and environment:
macOS 10.15.6, Chrome 84.0.4147.105 (though likely happens in all environments)

### Affects versions
2.19.x

### PR DESCRIPTION
**Pull request by leo-sakh**
#### What does this PR do?
Fixed #6204 

#### Who is reviewing it? 
@bennuttle 
@lavoywj 
@jMoneee 
@frnkshin 

#### Select relevant component teams: 
@codice/ui 

#### Ask 2 committers to review/merge the PR and tag them here.
@bdeining
@millerw8

#### How should this be tested?
See #6204 for steps to reproduce, these should be used to verify

#### What are the relevant tickets?
Fixes: #6204

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/enum/enum.tsx [hunk 1]
```diff
@@ -63,6 +67,7 @@ export default hot(module)(
         if (nextProps.value !== this.props.value) {
           this.state.propertyModel &&
             this.state.propertyModel.setValue([nextProps.value])
+          this.setState({ regionComponentKey: Math.random() }) //Forces re-render of the MarionetteRegionContainer
```
Comment (millerw8) [inline ui/packages/catalog-ui-search/src/main/webapp/react-component/enum/enum.tsx:70]: so we are using a random number just to cause the state to change here?  Really has no meaning just counting on it to be different?

Comment (leo-sakh) [reply ui/packages/catalog-ui-search/src/main/webapp/react-component/enum/enum.tsx:70]: Basically, yes, without it the component will not change so no re-render. But it's also a React component key so there is that.

**Comment by leo-sakh** [comment]
build now

**Comment by jMoneee** [comment]
Heroing

**Comment by jMoneee** [comment]
Hero Successful. Table Export window no longer flickers on change of input value.

**Pull request merged**